stdlib/template/autotools.py

```diff
@@ -0,0 +1,138 @@
+
+    **Patch**
+
+        This step is used to patch the downloaded source code. The default value is :py:func:`.patch_all` with no argument.
+
+    From now one, the current working directory is changed in favor of ``build_folder`` (which defaults to ``.``).
+    If the directory pointed by ``build_folder`` doesn't exist, it is created.
+
+    This is useful for the few ``configure`` scripts that don't work if they aren't executed in a standalone directory.
+
+    **Configure**
+
+        This step uses the ``configure`` script to configure the source code. The default value is :py:func:`.configure` with no argument.
+
+        When ``build_folder`` isn't ``.``, it is usually necessary to override this step with an other call to :py:func:`.configure` with
+        the argument ``binary='../configure'``.
+
+    **Compile**
+
+        This step compiles the source code. The default value is :py:func:`.make` with no argument.
+
+    **Check**
+
+        This step runs the unit and integration tests. The default value is :py:func:`.make` with the arguments ``'check'`` and ``fail_ok=True``.
+
+    **Install**
+
+        This step installs the software in the install cache. The default value is :py:func:`.make` with the arguments ``'install'`` and
+        ``f'DESTDIR={stdlib.build.current_build().install_cache}'``
+
+        If this step is overwritten, the ``DESTDIR`` variable is crucial and should be reused. Otherwise, and unless the ``configure`` script has been
+        configured to work without it, the installation can damage and overwrite parts of the host system.
+
+    **Split**
+
+        This step is still a work in progress. The default value is ``None``.
+
+    **Dependency Linking**
+
+        This step is still a work in progress. The default value is ``None``.
+    """
+    build = stdlib.build.current_build()
+
+    stdlib.log.ilog("Step 1/9: Fetch")
+    if fetch is not None:
+        fetch()
+
+    stdlib.log.ilog("Step 2/9: Extract")
+    if extract is not None:
+        extract()
+
+    stdlib.log.ilog("Step 3/9: Patch")
+    if patch is not None:
+        patch()
+
+    os.makedirs(build_folder, exist_ok=True)
+    with stdlib.pushd(build_folder):
+
+        print(f"OMG I'M IN {os.getcwd()}")
+        os.environ['DESTDIR'] = build.install_cache
+
+        stdlib.log.ilog("Step 4/9: Configure")
+        if configure is not None:
+            configure()
+
+        stdlib.log.ilog("Step 5/9: Compile")
+        if compile is not None:
+            compile()
+
+        stdlib.log.ilog("Step 6/9: Check")
+        if check is not None:
+            check()
+
+        stdlib.log.ilog("Step 7/9: Install")
+        if install is not None:
+            install()
+
+        stdlib.log.ilog("Step 8/9: Split")
+        if split is not None:
+            split()
+
+        stdlib.log.ilog("Step 9/9: Dependency Linking")
+        if deplinker is not None:
+            deplinker()
+
+    return []
```

Comment:
I think I need an explanation (and we probably need a doc comment) regarding why the empty list is returned here.

Reply:
That is temporary.

The return value should be the one of the splitter, but this part of Nbuild isn't revamped yet, so there is no return value so far.